src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -283,16 +325,26 @@ class HuiMapCard extends LitElement implements LovelaceCard {
       return;
     }
 
-    const bounds = this.Leaflet.latLngBounds(
-      this._mapItems ? this._mapItems.map((item) => item.getLatLng()) : []
-    );
+    const bounds = this.Leaflet.featureGroup(this._mapItems).getBounds();
     this._leafletMap.fitBounds(bounds.pad(0.5));
 
     if (zoom && this._leafletMap.getZoom() > zoom) {
       this._leafletMap.setZoom(zoom);
     }
   }
 
+  private _getColor(name: string) {
```

Comment:
This is only called with entityId, and so parameter should be entityId.

Reply:
I tried to keep this function more generic. input a name and if you request a color for the same name again, you get the same color. Independent what kind of name it is. 
But I have changed it.